src/value.c

```diff
@@ -259,6 +261,53 @@ const char *SIType_ToString(SIType t) {
 	}
 }
 
+void SIType_ToMultipleTypeString(SIType t, char **buf, size_t *bufferLen, size_t *bytesWritten) {
+	SIType remainingTypes;
+	SIType currentType;
+	// Allocate space
+	// Worst case: Len(SIType names) + 19*Len(", ") + Len("Or") = 177 + 38 + 2 = 217
+	if(*bufferLen - *bytesWritten < 256) {
+		*bufferLen += 256;
+		*buf = rm_realloc(*buf, sizeof(char) * *bufferLen);
+	}
+
+	// Remove from remainingTypes the types which are not used currently
+	remainingTypes = t & (~0x7E0);
+
```

Comment:
Thanks.

Reply:
Hi, sorry about this, but on a second thought let's keep this function generic, it shouldn't be aware about which SI_Types are actually in use and which aren't, let's take all types within the enum into consideration, there's no real harm in doing so.

Reply:
No problem. Change done.

1. SIType_ToString() was modified to support all types, because tostring() accepts Datetime, Duration types. Is that OK?
2. What do you think of replace:
```C
for(int i = 0; i < 18; i ++) {
    currentType = (1 << i);
 ```
By this:
```C
for(currentType = 1; currentType < SI_ALL; currentType = currentType << 1) {
```

Reply:
This alternative loop seems better, I don't expect it to overflow.

Reply:
The loop was updated. [30f2f42](https://github.com/RedisGraph/RedisGraph/pull/2691/commits/30f2f428c72572aa87fd945f293458b0f2cacb54)

Reply:
@AviAvni I think your code is more readable.
The second loop needs to check ```if (t & currentType == 0)```, change was added in [f4c0207](https://github.com/RedisGraph/RedisGraph/pull/2691/commits/f4c0207bfce62a3acdfd718325bbac836807d583)